Sources/LiveViewNative/Coordinators/LiveSessionCoordinator.swift

```diff
@@ -103,26 +128,40 @@ public class LiveSessionCoordinator<R: RootRegistry>: ObservableObject {
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] newView in
+                guard let self else { return }
+                guard let last = self.navigationPath.last else { return }
+                if let client = self.liveviewClient {
+                    last.coordinator.join(client, self.eventHandler, self.patchHandler)
+                }
+        }.store(in: &cancellables)
 
         $navigationPath.scan(([LiveNavigationEntry<R>](), [LiveNavigationEntry<R>]()), { ($0.1, $1) }).sink { [weak self] prev, next in
             guard let self else { return }
+            guard let client = liveviewClient else { return }
             Task {
-                try await prev.last?.coordinator.disconnect()
+                prev.last?.coordinator.disconnect()
                 if prev.count > next.count {
                     // back navigation (we could be going back multiple pages at once, so use `traverseTo` instead of `back`)
-                    let targetEntry = self.liveSocket!.getEntries()[next.count - 1]
-                    next.last?.coordinator.join(
-                        try await self.liveSocket!.traverseTo(targetEntry.id, next.last!.coordinator.liveChannel, nil)
-                    )
+                    var opts = NavActionOptions()
+                    opts.joinParams = .some([ "_interface": .object(object: LiveSessionParameters.platformParams)])
+                    let targetEntry = client.getEntries()[next.count - 1]
+                    let _ = try await client.traverseTo(targetEntry.id, opts)
                 } else if next.count > prev.count && prev.count > 0 {
                     // forward navigation (from `redirect` or `<NavigationLink>`)
-                    next.last?.coordinator.join(
-                        try await self.liveSocket!.navigate(next.last!.url.absoluteString, next.last!.coordinator.liveChannel, NavOptions(action: .push))
-                    )
+                    var opts = NavOptions()
+                    opts.joinParams = .some([ "_interface": .object(object: LiveSessionParameters.platformParams)])
+                    opts.action = .push
+                    let _ = try await client.navigate(next.last!.url.absoluteString, opts)
                 } else if next.count == prev.count {
-                    guard let liveChannel = try await self.liveSocket?.navigate(next.last!.url.absoluteString, next.last!.coordinator.liveChannel, NavOptions(action: .replace))
-                    else { return }
-                    next.last?.coordinator.join(liveChannel)
+                    // TODO: this will fire on a patch event! this should not fire on a patch event
+                    var opts = NavOptions()
+                    opts.joinParams = .some([ "_interface": .object(object: LiveSessionParameters.platformParams)])
+                    opts.action = .replace
+                    let _ = try await client.navigate(next.last!.url.absoluteString, opts)
```

Comment:
There is a problem here, core now runs it's own even loop and will take patch and redirect events, these are currently not handled if they are not triggered by a client action. Those events are all pushed through `SimpleEventHandler.viewReloadSubject`. We need to watch for events that are emitted by server-side events and rectify the navigation path. 

the other option is just to call the appropriate client navigation functions wherever we would previously just modify the navigation path.

Reply:
I'm not following the specific issue here. Can you provide more detail?

Reply:
previously the swiftui client listened on the live view channel for `live_reload` and other `push_navigation` type events, then it would modify the `LiveSessionCoordinator.NavigationPath` and all navigation would happen downstream of checking for changes in the `scan` block like we do here. 

Now that push navigation events are handled opaquely inside core, we need to update the `LiveSessionCoordinator.NavigationPath` in response to internal events. While the code provided here works, the `NavigationPath` should be controlled *by*  core, which is the opposite of how it works now.

Reply:
oh nice :)

Reply:
and yes, I agree